### ISSUE TITLE
Segfaults when running Hypotests (from ROOT?)

### ISSUE THREAD

**Issue by Dziesma** [histfitter/histfitter#213]
I encountered segfaults when running hypothesis tests in various environments. I initially encountered these errors with my own scripts, but they are also reproduced when running: `MySimpleChannelConfig.py` from the tutorial.

```
HistFitter.py -w -p analysis/tutorial/MySimpleChannelConfig.py
```

This is what I noticed re-running this job in various environments:

- **el9 (lxplus) with StatAnalysis,0.3.0 and 0.3.2:** gave segfaults for both frequentist and asymptotic fits
- **centos7 with StatAnalysis,0.3.2:** gave segfaults only for frequentist fits, assymptotic fits worked fine
- **centos7 LCG_105 (root v6.30) compiled locally:** _sometimes_ gave segfaults for frequentist fits, assymptotic fits worked fine. In this case the segfaults were unpredictable, when running identical jobs one after the other, sometimes one crashed and the other finished just fine.
- **centos7 LCG_104a (root v6.28.06) compiled locally:** worked as expected.

My suspicion is that some root version compatibility issue was at fault.

Attached is a log (including my workflow) reproducing this error on lxplus (el9), along with the modified (changing asymptotic to frequentist and setting nToys) `MySimpleChannelConfig.py` used. 

In summary, hypothesis tests, especially with frequentist calculators, give unexpected segfaults in LCG_105, and StatAnalysis,0.3.X .

[MySimpleChannelConfig.py.txt](https://github.com/histfitter/histfitter/files/14792286/MySimpleChannelConfig.py.txt)
[histfitter_log.txt](https://github.com/histfitter/histfitter/files/14792228/histfitter_log.txt)
